src/framework/audio/engine/internal/mixer.cpp

```
@@ -87,37 +87,38 @@ RetVal<MixerChannelPtr> Mixer::addChannel(const TrackId trackId, ITrackAudioInpu
                 source->setIsActive(false);
+            } else {
+                source->setIsActive(isActive());
+                source->seek(secsToMicrosecs(playbackPosition().time()));
             }
-            if (m_nonMutedTrackCount != 0) {
-                m_nonMutedTrackCount--;
-            }
-            return;
         }
+    });
 
```

Cppcheck (2.20.0)

[style] The function 'setInputParams' is never used.

(unusedFunction)